File: AN_2/SEM_2/MPP/csharp_projects/lab2/repository/RepositoryInterface.cs
namespace MPP_Csharp.repository;

public interface RepositoryInterface
{
    public interface IRepository<T>
    {
        void adauga(T entity); 
        void sterge(int id);
        void modifica(T entity);
        T findById(int id);
        List<T> findAll();


    }
}

File: AN_2/SEM_2/MPP/java/lab4/src/main/java/ro/mpp/labfx/repository/RepositoryInterface.java
package ro.mpp.labfx.repository;

import java.util.List;

public interface RepositoryInterface<ID,T> {
    void adauga(T entity);
    void sterge(ID id);
    void modifica(T entity);
    T findById(ID id);
    List<T> findAll();

}